map/map_loader/src/pointcloud_map_loader/selected_map_loader_module.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "selected_map_loader_module.hpp"
+namespace
+{
+autoware_map_msgs::msg::PointCloudMapMetaData createMetadata(
+  const std::map<std::string, PCDFileMetadata> & pcd_file_metadata_dict)
+{
+  autoware_map_msgs::msg::PointCloudMapMetaData metadata_msg;
+  metadata_msg.header.frame_id = "map";
+  metadata_msg.header.stamp = rclcpp::Clock().now();
+
+  for (const auto & ele : pcd_file_metadata_dict) {
+    std::string path = ele.first;
+    PCDFileMetadata metadata = ele.second;
+
+    // assume that the map ID = map path (for now)
+    std::string map_id = path;
+
+    autoware_map_msgs::msg::PointCloudMapCellMetaDataWithID cell_metadata_with_id;
+    cell_metadata_with_id.cell_id = map_id;
+    cell_metadata_with_id.metadata.min_x = metadata.min.x;
+    cell_metadata_with_id.metadata.min_y = metadata.min.y;
+    cell_metadata_with_id.metadata.max_x = metadata.max.x;
+    cell_metadata_with_id.metadata.max_y = metadata.max.y;
+
+    metadata_msg.metadata_list.push_back(cell_metadata_with_id);
+  }
+
+  return metadata_msg;
+}
+}  // namespace
+
+SelectedMapLoaderModule::SelectedMapLoaderModule(
+  rclcpp::Node * node, const std::map<std::string, PCDFileMetadata> & pcd_file_metadata_dict)
+: logger_(node->get_logger()), all_pcd_file_metadata_dict_(pcd_file_metadata_dict)
+{
+  get_selected_pcd_maps_service_ = node->create_service<GetSelectedPointCloudMap>(
+    "service/get_selected_pcd_map", std::bind(
+                                      &SelectedMapLoaderModule::onServiceGetSelectedPointCloudMap,
+                                      this, std::placeholders::_1, std::placeholders::_2));
+
+  // publish the map metadata
+  rclcpp::QoS durable_qos{1};
+  durable_qos.transient_local();
+  pub_metadata_ = node->create_publisher<autoware_map_msgs::msg::PointCloudMapMetaData>(
+    "output/pointcloud_map_metadata", durable_qos);
+  pub_metadata_->publish(createMetadata(all_pcd_file_metadata_dict_));
+}
+
+bool SelectedMapLoaderModule::onServiceGetSelectedPointCloudMap(
+  GetSelectedPointCloudMap::Request::SharedPtr req,
+  GetSelectedPointCloudMap::Response::SharedPtr res)
```

Comment:
Please make this a `const` function

Reply:
Thanks!
I added const in https://github.com/autowarefoundation/autoware.universe/pull/3286/commits/3b2d764a15d782ac5b822e27193c621832f4b1ba

And we need to change [onServiceGetPartialPointCloudMap](https://github.com/autowarefoundation/autoware.universe/blob/6b55253b86bdb943f5d9ee865919d679c05503aa/map/map_loader/src/pointcloud_map_loader/partial_map_loader_module.cpp#L53-L60) and [onServiceGetDifferentialPointCloudMa](https://github.com/autowarefoundation/autoware.universe/blob/6b55253b86bdb943f5d9ee865919d679c05503aa/map/map_loader/src/pointcloud_map_loader/differential_map_loader_module.cpp#L66-L75) into const in another PR.